quizzes/Quiz-9-Concurrency-Control.md

```diff
@@ -0,0 +1,107 @@
+Q1 - Which of the following locks the item from change but not from read?
+
+- A. Exclusive lock
+- B. Explicit lock
+- C. Implicit lock
+- D. Shared lock
+
+Answer - D. Shared lock
+
+---
+
+Q2 - Which of the following locks the item from access of any type?
+
+- A. Implicit lock
+- B. Exclusive lock
+- C. Shared lock
+- D. Explicit lock
+
+Answer - B. Exclusive lock
+
+---
+
+Q3 - The size of a lock is called:
+
+- A. implicit lock.
+- B. lock granularity.
+- C. shared lock.
+- D. exclusive lock.
+
+Answer - B. lock granularity.
+
+---
+
+Q4 - Which refers to a stalemate situation due to which no further progress is possible as computer await response of each other:
+
+- A. Recovery
+- B. Deadlock
+- C. Backup
+- D. Concurrency
+
+Answer - B. Deadlock
+
+---
+
+Q5 - Isolation of the transactions is ensured by
+
+- A. Transaction management
+- B. Concurrency control
+- C. Application programmer
+- D. Recovery management
+
+Answer - B. Concurrency control
+
+---
+
+Q6 - A lock that allows concurrent transactions to access different rows of the same table is known as a
+
+- A. Database-level lock
+- B. Page-level lock
+- C. Table-level lock
+- D. Row-level lock
+
+Answer - D. Row-level lock
+
+---
+
+Q7 - Which of the following protocols ensures conflict serializability and safety from deadlocks?
+
+- A. Two-phase locking protocol
+- B. Time-stamp ordering protocol
+- C. Graph based protocol
+- D. None of the mentioned
+
+Answer - B. Time-stamp ordering protocol
+
+---
+
+Q8 - Which of the following is a procedure for acquiring the necessary locks for a transaction where all necessary locks are acquired before any are released?
+
+- A. Record controller
+- B. Two phase lock
+- C. Exclusive lock
+- D. Authorization rule
+
+Answer - B. Two phase lock
+
+---
+
+Q9 - A deadlock exists in the system if and only if the wait-for graph contains a(an) ____
```

Comment:
Answer of Q9 need check

Reply:
what seems to be the problem? checked it and it is coreect

Reply:
Yeah, my bad. That should be Cycle.